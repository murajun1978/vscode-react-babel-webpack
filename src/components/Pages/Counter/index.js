import React, { useReducer } from 'react';
import Button from '../../Atoms/Button';
import { reducer, initialState } from '../../../reducers/counter';
import counterActions from '../../../actions/counter';
import './index.scss';

const Counter = () => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const actions = counterActions(dispatch);

  return (
    <div className="counter-container">
      <div>
        <p className="counter-label" data-testid="counter">
          {state.count}
        </p>
      </div>
      <Button onClick={actions.reset}>Reset</Button>
      <Button onClick={actions.increment}>+</Button>
      <Button onClick={actions.decrement}>-</Button>
    </div>
  );
};

export default Counter;
