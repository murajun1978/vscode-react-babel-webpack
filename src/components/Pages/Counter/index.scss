.counter-container {
  .counter-label {
    color: red;
  }
}
