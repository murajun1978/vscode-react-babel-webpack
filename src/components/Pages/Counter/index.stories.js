import React from 'react';
import { storiesOf } from '@storybook/react';
import base from 'paths.macro';
import Counter from './';

const storyPath = base.replace('/src/components', '');

storiesOf(`${storyPath}`, module).add('Counter', () => <Counter />);
