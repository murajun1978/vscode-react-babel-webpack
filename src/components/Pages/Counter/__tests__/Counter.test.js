import React from 'react';
import { render } from 'react-testing-library';
import Counter from '../';

test('Counter', () => {
  const { container } = render(<Counter />);
  expect(container).toMatchSnapshot();
});
