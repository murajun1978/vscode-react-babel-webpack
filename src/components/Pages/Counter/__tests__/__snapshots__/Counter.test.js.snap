// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter 1`] = `
<div>
  <div
    class="counter-container"
  >
    <div>
      <p
        class="counter-label"
        data-testid="counter"
      >
        0
      </p>
    </div>
    <button>
      Reset
    </button>
    <button>
      +
    </button>
    <button>
      -
    </button>
  </div>
</div>
`;
